src/util.ts

```diff
@@ -230,11 +230,12 @@ export async function executeAsTask(name: string, command: string, args?: string
 // executes a callback and shows a 'busy' progress bar during the execution
 // synchronous callbacks are converted to async to properly render the progress bar
 // default location is in the help pages tree view
-export async function doWithProgress<T>(cb: () => T | Promise<T>, location: string | vscode.ProgressLocation = 'rHelpPages'): Promise<T> {
+export async function doWithProgress<T>(cb: () => T | Promise<T>, location: string | vscode.ProgressLocation = 'rHelpPages', title?: string): Promise<T> {
     const location2 = (typeof location === 'string' ? { viewId: location } : location);
     const options: vscode.ProgressOptions = {
         location: location2,
-        cancellable: false
+        cancellable: false,
```

Comment:
Is it possible to make it cancellable by killing the spawned process?

If a Rmd document contains time-consuming code, it might take a long time to render. A cancellable progress might make sense in this case.

Reply:
Good idea, just pushed an update to implement this

Reply:
I test with a computing heavy Rmd and it looks like the progress only shows quite shortly during the spawning rather than rendering the Rmd. The shiny app already displays a "Loading" during the rendering. Then the cancel button is only useful if the spawning takes a long time.

Reply:
Right. I guess it is useful if people use the `xaringan::infinite_moon_runner`, as this seems to actually knit the documents, as opposed to using a shiny backend.

Reply:
Having said that, I'm not 100% sold on having two preview 'engines'. I put it in initially as `infinite_moon_runner` seemed to not run into the issues of `rmarkdown::run`, such as the loss of text decoration.